uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerHost.java

```diff
@@ -293,15 +327,111 @@ private final String basicButtonHandler(@NonNull final String resourceButtonId,
+        List<String> accounts = new ArrayList<>();
+        final String accountName = getDialogBoxText(dialogBox, textId);
+        if (accountName != null) {
+            accounts.add(accountName);
+        }
+        if (expectingMultipleAccounts) {
+            final String accountName2 = getDialogBoxText(dialogBox, textId);
+            if (accountName2 != null)
+                accounts.add(accountName2);
+        }
+        return accounts;
+    }
+
+    /**
+     * Removes the added account
+     */
+    public void removeAccount(@NonNull final String username) {
+        final UiObject removeAccount = UiAutomatorUtils.obtainUiObjectWithResourceId(
+                "com.microsoft.identity.testuserapp:id/buttonRemoveAccount"
+        );
+        UiAutomatorUtils.obtainChildInScrollable("someone@contoso.com");
+        UiAutomatorUtils.handleInput("com.microsoft.identity.testuserapp:id/editTextUsername", username);
+        try {
+            removeAccount.click();
+        } catch (final UiObjectNotFoundException e) {
+            throw new AssertionError(e);
+        }
+    }
+
+    /**
+     * Decode the sso token and look for the requested nonce
+     */
+    public void confirmNoncePresent() {
+        // fill the nonce
+        UiAutomatorUtils.obtainChildInScrollable("nonce");
+        UiAutomatorUtils.handleInput("com.microsoft.identity.testuserapp:id/editTextNonce", "testNonce");
+
+        // click on sso token
+        UiAutomatorUtils.handleButtonClick("com.microsoft.identity.testuserapp:id/buttonGetSsoToken");
+        final UiObject ssoToken = UiAutomatorUtils.obtainUiObjectWithResourceId(
+                "com.microsoft.identity.testuserapp:id/sso_token"
+        );
+
+        try {
+            Assert.assertNotEquals(ssoToken.getText(), "");
+            UiAutomatorUtils.handleButtonClick("com.microsoft.identity.testuserapp:id/btnCookieDecode");
+            final UiObject decodeObj = UiAutomatorUtils.obtainUiObjectWithResourceId(
+                    "com.microsoft.identity.testuserapp:id/sso_token"
+            );
+            Assert.assertNotEquals(decodeObj.getText(), "");
+
+            final String decodedTxt = decodeObj.getText();
+            // confirm decoded string has the nonce
+            if (decodedTxt.contains("request_nonce")) {
+                final String[] str = decodedTxt.split("request_nonce => ");
+                if (str.length > 1) {
+                    Assert.assertEquals(str[1].trim(), "testNonce");
+                } else {
+                    Assert.fail("decoded token does not contain correct nonce");
+                }
+            } else {
+                Assert.fail("decoded token does not contain correct nonce");
+            }
+        } catch (final UiObjectNotFoundException e) {
+            throw new AssertionError(e);
+        }
+    }
+
+    /**
+     * Confirm that the calling app is not verified
+     */
+    public void confirmCallingAppNotVerified() {
```

Comment:
It seems we're trying to verify an error scenario. When is someone supposed to call this method?

Reply:
This is being called in the test case 1600567 https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1600567

Here we perform multiple actions like getAccounts, removeAccount, getFlights etc and verify that the dialog always has the "calling App not verified" in the dialog.